src/search.py

```diff
@@ -0,0 +1,62 @@
+from collections import deque
+import st
+
+def LCP(a: str, b: str) -> int:
+    """
+    Finds the longest common prefix of a and b
+    
+    returns: 
+    If there is no prefix returns 0 otherwise returns length of prefix
+    """
+    i = 0
+    while(i < len(a) and i < len(b)):
+        if a[i] != b[i]:
+            return i
+        i+=1
+    return i
+
+def search(root: st.Node, pattern: str, string: str):
+    """
+    Looks for a pattern in a string using a suffix tree of string.
+    """
+    #Avoid copies while slicing with memoryviews
+
+    patternView = memoryview(pattern.encode())
+    stringView = memoryview(string.encode())
+
+    currNode = root.child
+    
+    patternProcessed = 0
+
+    #Find pattern (I)
+    queue = deque()
+    
+    while(True):
+
+        prefixLength = LCP(patternView[patternProcessed:], stringView[currNode.start:currNode.end])
```

Comment:
Does this match the type annotation for `LCP()`? It expects `str` arguments, right?